src/cmake-tools.ts

```diff
@@ -243,17 +250,49 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       vscode.window.showErrorMessage(localize('no.source.directory.found', 'You do not have a source directory open'));
       break;
     case CMakePreconditionProblems.MissingCMakeListsFile:
-      const quickStart = localize('quickstart.cmake.project', 'Quickstart a new CMake project');
-      const changeSetting = localize('edit.setting', 'Edit the \'cmake.sourceDirectory\' setting');
-      const result = await vscode.window.showErrorMessage(
-            localize('missing.cmakelists', 'CMakeLists.txt was not found in the root of the folder \'{0}\'', this.folderName), quickStart, changeSetting);
-      if (result === quickStart) {
-        vscode.commands.executeCommand('cmake.quickStart');
-      } else if (result === changeSetting) {
-        vscode.commands.executeCommand('workbench.action.openSettings');
+      if (!this.workspaceContext.state.ignoreCMakeListsMissing) {
+        const quickStart = localize('quickstart.cmake.project', 'Create');
+        const changeSetting = localize('edit.setting', 'Locate');
+        const ignoreCMakeListsMissing = localize('ignore.activation', "Don't show again");
+        const result = await vscode.window.showErrorMessage(
+            localize('missing.cmakelists', 'CMakeLists.txt was not found in the root of the folder \'{0}\'. How would you like to proceed?', this.folderName),
+            quickStart, changeSetting, ignoreCMakeListsMissing);
+        if (result === quickStart) {
+          vscode.commands.executeCommand('cmake.quickStart');
+        } else if (result === changeSetting) {
+          const openOpts: vscode.OpenDialogOptions = {
+            canSelectMany: false,
+            defaultUri: this.folder.uri,
+            filters: {"CMake files": ["txt"], "All files": ["*"]},
+            openLabel: "Load",
+          };
+          const cmakeListsFile = await vscode.window.showOpenDialog(openOpts);
+          if (cmakeListsFile) {
```

Comment:
If no `cmakeListsFile` is selected, then we need to disable the extension features.

Reply:
I can do that. Originally I thought that we disable the features only if the user chooses "ignore".